### ISSUE TITLE
No sound on latest ArchLinux kernel

### ISSUE THREAD

**Issue by iTrooz** [codepayne/linux-sound-huawei#31]
Hello !
Since the release of the 6.7 kernel, which should have the patch from this repository, I tried using it from the official Arch Linux repositories, but I don't have any sound (KDE doesn't detect any input/output device). I'm not sure how to debug this, could someone help ?

I remember that I was able to make https://github.com/codepayne/pop-os-linux work at some point (EDIT: I just remembered I had to patch it to add my model, sorry)

- `uname -a`: `Linux arch-3306 6.7.0-arch3-1 #1 SMP PREEMPT_DYNAMIC Sat, 13 Jan 2024 14:37:14 +0000 x86_64 GNU/Linux`
- `/proc/config.gz` content:  [config.txt](https://github.com/codepayne/linux-sound-huawei/files/13954142/config.txt)
- Relevant `dmidecode` part:
```
System Information
        Manufacturer: HUAWEI
        Product Name: HVY-WXX9
        Version: M1010
        Serial Number: [hidden]
        UUID: [hidden]
        Wake-up Type: Other
        SKU Number: C100
        Family: MateBook D
```
- [Alsamixer screenshot](https://github.com/codepayne/linux-sound-huawei/assets/42669835/f67cfe99-abf8-48b0-a522-0cd59920bdd4)

**Comment by iTrooz** [comment]
dmesg logs from a fresh boot + `aplay sample-12s.wav`: [dmesg.txt](https://github.com/codepayne/linux-sound-huawei/files/13954258/dmesg.txt)

**Comment by andreaconsole** [comment]
If you have a HUAWEI BOM-WXX9 M1010, you shoud check this:
https://github.com/codepayne/linux-sound-huawei/issues/28

**Comment by iTrooz** [comment]
Unfortunately I have a HVY-WXX9 M1010, and the sound card doesn't even show up in alsamixer, so I don't think your linked issue would help

**Comment by saber716rus** [comment]
> Unfortunately I have a HVY-WXX9 M1010, and the sound card doesn't even show up in alsamixer, so I don't think your linked issue would help

@andreaconsole Most likely, there is no necessary module in the kernel.

@iTrooz do it and put the log here.
`  cd ~/
  wget http://www.alsa-project.org/alsa-info.sh -O alsa-info.sh && bash alsa-info.sh`

**Comment by iTrooz** [comment]
http://alsa-project.org/db/?f=a16d221230323a15882f4e7da449dc4bf235ddb3

**Comment by saber716rus** [comment]
> http://alsa-project.org/db/?f=a16d221230323a15882f4e7da449dc4bf235ddb3

Is your aur connected?

**Comment by codepayne** [comment]
@iTrooz 

```
[   10.398143] acp_mach acp3x-es83xx: this system has a ES83xx codec defined in ACPI, but the driver doesn't have this system registered in DMI table
[   10.398147] acp_mach acp3x-es83xx: Cannot probe card (acp3x-es83xx): -19
```
Your laptop model is not supported for some reason. Seems I didn't add support for it in the upstream kernel driver, but had it enabled in my own kernel repos. Will fix this but it will take time to reach an official kernel release.

**Comment by iTrooz** [comment]
Okay ! Thanks a lot for your work

**Comment by codepayne** [comment]
@iTrooz The fix is upstream

https://lore.kernel.org/alsa-devel/170656972577.167619.10975309830352021265.b4-ty@kernel.org/T/#t

**Comment by iTrooz** [comment]
Thanks !

**Comment by AlexUkr73** [comment]
@codepayne 

Good day,

I have the same issue, but in dmidecode part I have

Product Name: HVY-WXX9-PCB-B1

I've tried debian kernels on MX-Linux:
 6.8.9.1-liquorix
 6.7.12-1-liquorix
 6.5.0-ahs (Advansed Hardware support)- most resent.
 

Issue does not reproduce from Live run of Ubuntu 24.04 (kernel 6.8.0.13)

Could you advice any workarounds, please?

Update (after 2 days):
Please disregard - now sound is available for me with 6.8.1 kernel compiled from source/
